### PR TITLE
test(ci): add azp image builds

### PR DESCRIPTION
**Pull request by matmerr**
# Description

Need to be able to build images in azp as well.

## Related Issue

If this pull request is related to any issue, please mention it here. Additionally, make sure that the issue is assigned to you before submitting this pull request.

## Checklist

- [ ] I have read the [contributing documentation](https://retina.sh/docs/contributing).
- [ ] I signed and signed-off the commits (`git commit -S -s ...`). See [this documentation](https://docs.github.com/en/authentication/managing-commit-signature-verification/about-commit-signature-verification) on signing commits.
- [ ] I have correctly attributed the author(s) of the code.
- [ ] I have tested the changes locally.
- [ ] I have followed the project's style guidelines.
- [ ] I have updated the documentation, if necessary.
- [ ] I have added tests, if applicable.

## Screenshots (if applicable) or Testing Completed

Please add any relevant screenshots or GIFs to showcase the changes made.

## Additional Notes

Add any additional notes or context about the pull request here.

---

Please refer to the [CONTRIBUTING.md](../CONTRIBUTING.md) file for more information on how to contribute to this project.

### CODE REVIEW

**Comment by matmerr** [comment]
needs operator

File: .pipelines/cg-pipeline.yaml [hunk 1]
```diff
@@ -43,4 +43,100 @@ stages:
...
+          - script: |
+              set -euo pipefail
+              echo "VERSION=$(make version)"
+              export VERSION=$(make version)
+              mkdir -p ./output/images/$(platform)/$(arch)/$(year)
+              make retina-image-win \
+                TARGET=final \
+                WINDOWS_YEARS=$(year) \
+                TAG=$(make version) \
+                BUILDX_ACTION="-o type=docker,dest=./output/images/$(platform)/$(arch)/$(year)/retina-agent-$VERSION.tar"
```
Comment (alam-tahmid) [inline .pipelines/cg-pipeline.yaml:130]: can we add os for the images keeping the parity with the gh images. `retina-agent-$VERSION-linux-$(arch).tar`

File: .pipelines/cg-pipeline.yaml [hunk 2]
```diff
@@ -43,4 +43,100 @@ stages:
...
+              echo "VERSION=$(make version)"
+              export VERSION=$(make version)
+              mkdir -p ./output/images/$(platform)/$(arch)/$(year)
+              make retina-image \
+                AGENT_TARGETS=$(target) \
+                TAG=$(make version) \
+                IMAGE_NAMESPACE=retina \
+                PLATFORM=$(platform)/$(arch) \
+                IMAGE_REGISTRY=ghcr.io/microsoft  \
+                BUILDX_ACTION="-o type=docker,dest=./output/images/$(platform)/$(arch)/retina-$(target)-$VERSION.tar"
```
Comment (alam-tahmid) [inline .pipelines/cg-pipeline.yaml:88]: can we add os for the images keeping the parity with the gh images. retina-agent-$VERSION-linux-$(arch).tar

File: .pipelines/cg-pipeline.yaml [hunk 3]
```diff
@@ -43,4 +43,105 @@ stages:
...
+              platform: "linux"
+              arch: "arm64"
+              target: "init"
+
+            init-linux-amd64:
+              platform: "linux"
+              arch: "amd64"
+              target: "init"
+            
+            operator-linux-amd64:
```
Comment (matmerr) [inline .pipelines/cg-pipeline.yaml:74]: in the github actions there's a separate job, maybe follow that if agent/operator need separate matrices for makefile commands, if that's easier

https://github.com/microsoft/retina/blob/main/.github/workflows/images.yaml#L109

File: .pipelines/cg-pipeline.yaml [hunk 4]
```diff
@@ -2,12 +2,12 @@ pr: none
...
-  branches:
-    include:
-    - main
-    - vakr/newadocg
+  - cron: "0 0 * * *"
+    displayName: Daily midnight Retina OSS CG pipeline
+    branches:
+      include:
+        - main
+        - vakr/newadocg
```
Comment (rbtr) [inline .pipelines/cg-pipeline.yaml:10]: can we delete this branch trigger

Comment (alam-tahmid) [reply .pipelines/cg-pipeline.yaml]: removed it @rbtr

**Review by alam-tahmid — DISMISSED** [review]

**Review by jimassa — DISMISSED** [review]

**Review by jimassa — DISMISSED** [review]

**Review by jimassa — APPROVED** [review]

**Pull request merged**